### DISCUSSION TITLE
Hook to execute an action on form completion, salesforce, and read only questions. Follow up with Amanda (111884)

### DISCUSSION THREAD

**Discussion by davidchoy** [impress-org/givewp#6148] [Developer Q&A] [answered]
Hi Jason + GiveWP team!

1. I asked Amanda over in sales/customer service, "After a donation is complete, can one call a script and do something else on demand?" She let me know, "Probably possible, yes. There are several hooks that they could use, and a well-formed question over on our Github discussions page would likely get an energetic explanation from Jason and the dev team." Can you point me in the right direction?

2. I also asked if transactions imported could be sent over to sales force. Amanda let me know your team is considering this. That said, if there is a hook that I could use to run code after a transaction is complete.... is that hook "run" for each imported transaction record? If so, could I then write code to send imported transaction data to salesforce myself? Do you have an example of what fields are sent, by default, to salesforce when a transaction is complete the regular way?

3. I also asked, "Can one programmatically create a donation manually via the GiveWP API? Example: completing a donation form once a Gravity Form entry is received." She let me know, "No, our API is read-only." Just double checking that makes sense with you — does it?

Thanks!
David F. Choy

**Comment by kjohnson** [comment, marked as answer]
I would probably suggest the `give_insert_payment` hook for custom code after a donation is created.

See `give/includes/payments/functions.php`

```php
/**
 * Fires while inserting payments.
 *
 * @since 1.0
 *
 * @param int   $payment_id   The payment ID.
 * @param array $payment_data Arguments passed.
 */
do_action( 'give_insert_payment', $payment->ID, $payment_data );
```

As for programmatically creating donations, I would suggest the `give_insert_payment( $paymentData )` function. There is not a provided REST API Endpoint for creating a payment, but the function is public access.

```php
$paymentData = [
  //
];
give_insert_payment( $paymentData );
```

**Comment by kjohnson** [comment]
In regards to Salesforce we recently released an official [Salesforce add-on for GiveWP](https://givewp.com/addons/salesforce/).

Otherwise, you could use the `give_insert_payment` action and custom code to push data to Salesforce.
